dll/win32/shell32/CDefView.cpp

```diff
@@ -3960,6 +3963,9 @@ HRESULT STDMETHODCALLTYPE CDefView::SetCallback(IShellFolderViewCB  *new_cb, ISh
         *old_cb = m_pShellFolderViewCB.Detach();
 
     m_pShellFolderViewCB = new_cb;
+    m_pFolderFilter = NULL;
```

Comment:
If `new_cb` is `NULL`, Should we also clear this `m_pFolderFilter` ?

Reply:
Is this relation between `IFolderFilter` and `IShellFolderViewCB` a ReactOS specific thing?
The documentation for `IFolderFilter` seems to suggest that the client has to set the filter via the `IFolderFilterSite` in response to the `BFFM_IUNKNOWN` message, instead of the host querying for it that is happening here.

Reply:
In my testing, on both NT5 and NT6, if you call SetCallback it will QueryInterface you for a new IFolderFilter. This makes sense because any dynamic state inside the object has to be in IShellFolderViewCB, it cannot be in IShellFolder because someone can call CreateViewObject on an IShellFolder multiple times and each returned view must be its own entity.

As an aside, Microsofts SetCallback does not follow the documentation, their ppOldCB is not optional.

Finally, I don't expect SetCallback to be called by anyone other than from inside SHCreateShellFolderView which we already do.

Reply:
> Is this relation between IFolderFilter and IShellFolderViewCB a ReactOS specific thing?

No, I have a simple IShellFolder & IShellFolderViewCB implementation that debug prints out stuff for testing on Windows. DefView asks for IFolderFilter via QueryInterface when we are talking about IShellFolderViewCB.

> The documentation for `IFolderFilter` seems to suggest that the client has to set the filter via the `IFolderFilterSite` in response to the `BFFM_IUNKNOWN` message, instead of the host querying for it that is happening here.

But that is for `SHBrowseForFolder`. I do plan to implement the folder for `SHBrowseForFolder` but it has nothing to do with this PR.

Reply:
Thanks for elaborating.
It would be nice to have this info somewhere in the repo, maybe this testproject can be dropped somewhere?
or a .md with some stuff like this.